txnsync/outgoing_test.go

```diff
@@ -0,0 +1,294 @@
+
+	err := asyncEncoder.asyncEncodeAndSend(nil)
+
+	a.Nil(err)
+	a.True(called)
+
+}
+
+func TestAssemblePeerMessage(t *testing.T) {
+
+	a := require.New(t)
+
+	s := syncState{
+		service:                    nil,
+		log:                        mockAsyncLogger{},
+		node:                       mockAsyncNodeConnector{},
+		isRelay:                    false,
+		clock:                      timers.MakeMonotonicClock(time.Now()),
+		config:                     config.Local{},
+		threadpool:                 nil,
+		genesisID:                  "",
+		genesisHash:                crypto.Digest{},
+		lastBeta:                   0,
+		round:                      42,
+		fetchTransactions:          false,
+		scheduler:                  peerScheduler{},
+		interruptablePeers:         nil,
+		interruptablePeersMap:      nil,
+		incomingMessagesQ:          incomingMessageQueue{},
+		outgoingMessagesCallbackCh: nil,
+		nextOffsetRollingCh:        nil,
+		requestsOffset:             0,
+		lastBloomFilter:            bloomFilter{},
+		transactionPoolFull:        false,
+		messageSendWaitGroup:       sync.WaitGroup{},
+		xorBuilder:                 bloom.XorBuilder{},
+	}
+
+	s.profiler = makeProfiler(1*time.Millisecond, s.clock, s.log, 1*time.Millisecond)
+
+	s.isRelay = false
+
+	peer := Peer{
+		networkPeer:                        nil,
+		isOutgoing:                         false,
+		state:                              0,
+		lastRound:                          0,
+		incomingMessages:                   messageOrderingHeap{},
+		nextReceivedMessageSeq:             0,
+		recentIncomingBloomFilters:         nil,
+		recentSentTransactions:             nil,
+		recentSentTransactionsRound:        0,
+		requestedTransactionsModulator:     0,
+		requestedTransactionsOffset:        0,
+		lastSentMessageSequenceNumber:      0,
+		lastSentMessageRound:               0,
+		lastSentMessageTimestamp:           0,
+		lastSentMessageSize:                0,
+		lastSentBloomFilter:                bloomFilter{},
+		lastConfirmedMessageSeqReceived:    0,
+		lastReceivedMessageLocalRound:      0,
+		lastReceivedMessageTimestamp:       0,
+		lastReceivedMessageSize:            0,
+		lastReceivedMessageNextMsgMinDelay: 0,
+		dataExchangeRate:                   0,
+		localTransactionsModulator:         0,
+		localTransactionsBaseOffset:        0,
+		lastTransactionSelectionTracker:    nil,
+		nextStateTimestamp:                 0,
+		messageSeriesPendingTransactions:   nil,
+		transactionPoolAckCh:               nil,
+		transactionPoolAckMessages:         nil,
+		lastSelectedTransactionsCount:      0,
+	}
+
+	pendingTransactions := pendingTransactionGroupsSnapshot{
+		pendingTransactionsGroups:           nil,
+		latestLocallyOriginatedGroupCounter: 0,
+	}
+
+	peer.setLocalRequestParams(111, 222)
+	peer.lastReceivedMessageTimestamp = 100
+	peer.lastReceivedMessageLocalRound = s.round
+
+	s.isRelay = true
+
+	metaMessage := s.assemblePeerMessage(&peer, &pendingTransactions)
```

Comment:
missing testing for some of the message options:
- messageConstBloomFilter
- messageConstTransactions